os/dutils.sh

```diff
@@ -57,6 +60,63 @@ function FIND_DBGBINFILE()
 	KERNEL_DBG_BIN="tinyara$EXEEXT"
 }
 
+function TRAP_MENU()
+{
+	echo =============================================================================
+	echo "  \"Select binary to debug crash with TRAP tool\""
+	echo =============================================================================
+	echo "  \"1. Default binary from previous build at build/output/bin location\""
+	echo "  \"2. Custom binary at any location\""
+	echo "  \"x. Exit\""
+	echo =============================================================================
+
+	read SELECT_BIN
+	case ${SELECT_BIN,,} in
+		1|default)
+			if [ -z "$(ls -A $BINDIR)" ]; then
+				echo "No output file, Build the code and run TRAP again."
+				exit 1
+			fi
+			echo "Enter the crash log file name: (ex: ../tools/trap/testlogs)"
+			read LOG_FILE
+			while [ ! -f ${LOG_FILE} ]; do
```

Comment:
Why do you use while statement to check the file? If the file is not present, it will go into an infinite loop. I think you need to check with if statement. And if the file is not present, you need to exit or go to the previous menu, so that user can re-enter the file name or select another option.

Reply:
Hi,
It does not enter an infinite loop. Also, it is checked again in parser tool. In fact it goes back to the previous menu
```
Enter the crash log file name: (ex: ../tools/trap/testlogs)

make: Entering directory '/home/vidisha/tizenRT/os/tools'
chmod 755 ../makeinfo.sh
make: Leaving directory '/home/vidisha/tizenRT/os/tools'
Executing: ramdumpParser.py -t  -e tinyara.axf
-e does not exist. Please provide the proper path for log_file...
======================================================
  "Select Option"
======================================================
  "1. Run Toolchain Command"
  "2. Run TRAP Debug tool"
  "h. Help"
  "x. Exit"
======================================================
```